### PR TITLE
Separate Dropdown and Autocomplete data

### PR DESCRIPTION
**Pull request by un-tone**
For https://github.com/StarcounterSamples/KitchenSink/issues/86

### CODE REVIEW

File: src/KitchenSink/BreadcrumbPage.json.cs [hunk 1]
```diff
@@ -22,13 +22,8 @@ protected override void OnData()
...
 
-            if (BreadcrumbTestData.Exists())
-            {
-                Db.Transact(() => { BreadcrumbTestData.DeleteAll(); });
-            }
-            BreadcrumbTestData.Create();
-
-            var treeItem = Db.SQL<TreeItem>("SELECT i FROM TreeItem i WHERE Name = ?", "Milk").First;
+            var treeItem = Db.SQL<TreeItem>("SELECT i FROM TreeItem i WHERE Name = ?", "Milk").First
+                ?? Db.SQL<TreeItem>("SELECT i FROM TreeItem i FETCH ?", 1).First;
```
Comment (warpech) [inline src/KitchenSink/BreadcrumbPage.json.cs]: Why is line 26 needed?

Comment (warpech) [inline src/KitchenSink/BreadcrumbPage.json.cs]: Why is line 26 needed?

Comment (un-tone) [inline src/KitchenSink/BreadcrumbPage.json.cs]: If a user changes the item with name "Milk" to any other name (e.g. "Milk2") the query `Db.SQL<TreeItem>("SELECT i FROM TreeItem i WHERE Name = ?", "Milk").First` will return `null` and in the following code the NullReferenceException will be thrown.
Doing this query `Db.SQL<TreeItem>("SELECT i FROM TreeItem i FETCH ?", 1).First` eliminates it

Comment (warpech) [inline src/KitchenSink/BreadcrumbPage.json.cs]: Good catch! But still if I change the name to "Milk2" and reload the page, I am getting null reference exception. Can you reproduce that?

Comment (un-tone) [inline src/KitchenSink/BreadcrumbPage.json.cs]: Sure, thanks. I missed a filtering by Parent field. Fixed.

File: src/KitchenSink/BreadcrumbPage.json.cs [hunk 2]
```diff
@@ -22,13 +22,8 @@ protected override void OnData()
...
-            if (BreadcrumbTestData.Exists())
-            {
-                Db.Transact(() => { BreadcrumbTestData.DeleteAll(); });
-            }
-            BreadcrumbTestData.Create();
-
-            var treeItem = Db.SQL<TreeItem>("SELECT i FROM TreeItem i WHERE Name = ?", "Milk").First;
+            var treeItem = Db.SQL<TreeItem>("SELECT i FROM TreeItem i WHERE Name = ?", "Milk").First
+                ?? Db.SQL<TreeItem>("SELECT i FROM TreeItem i WHERE Parent IS NOT NULL FETCH ?", 1).First;
             SetActiveItem(treeItem);
```
Comment (warpech) [inline src/KitchenSink/BreadcrumbPage.json.cs]: Now it works good, but maybe let's forget about "Milk" and simplify it by just calling:

```
var treeItem = Db.SQL<TreeItem>("SELECT i FROM TreeItem i WHERE Parent IS NOT NULL FETCH ?", 1).First;
```

Comment (un-tone) [inline src/KitchenSink/BreadcrumbPage.json.cs]: OK, removed, please check.

**Comment by KmaTTi89** [comment]
There is a problem while merging with KitchenSink.csproj

**Comment by un-tone** [comment]
@KmaTTi89, what is the problem? Could you say how can I check it?

**Comment by KmaTTi89** [comment]
@un-tone The problem is, I guess, your changes are based on dev branch and my current branch where i am testing has a lot of changes.
Could you try to merge your branch with my: new_tests_project

**Comment by un-tone** [comment]
@KmaTTi89, I fixed the formatting of the KitchenSink.csproj, please check

**Comment by warpech** [comment]
@KmaTTi89 I think that we need to decide to one target branch. Would you like this to be merged to `develop` or `new_tests_project`?

**Comment by warpech** [comment]
I think that the correct flow is to merge to `develop` and then @KmaTTi89 solves conflicts in his branch since he owns that branch.

**Comment by warpech** [comment]
Ok, then it is fine by me!

@un-tone maybe you can propose strategy to close this PR. Since @KmaTTi89 has merged from one feature branch to another feature branch, it should be enough to change the PR target from `develop` to `new_tests_project` to merge without conflicts.

**Comment by un-tone** [comment]
@warpech, in a standard case I would target these PRs to `develop`. But since these issues are related directly to the test covering issues I assume it more correct to target the PRs to the `new_tests_project` branch.

**Comment by un-tone** [comment]
Please confirm re-targeting of the PR.

**Comment by warpech** [comment]
I confirm.

**Comment by un-tone** [comment]
Closing. Changes from the `feature/Separate-Dropdown-Autocomplete-data` branch is already merged with `new_tests_project`

**Pull request closed without merging**